pyqtgraph/parametertree/ParameterItem.py

```diff
@@ -158,13 +158,20 @@ def nameChanged(self, param, name):
 
     def titleChanged(self):
         # called when the user-visble title has changed (either opts['title'], or name if title is None)
-        self.setText(0, self.param.title())
-        fm = QtGui.QFontMetrics(self.font(0))
-        textFlags = QtCore.Qt.TextFlag.TextSingleLine
-        size = fm.size(textFlags, self.text(0))
-        size.setHeight(int(size.height() * 1.35))
-        size.setWidth(int(size.width() * 1.15))
-        self.setSizeHint(0, size)
+
+        # when changing from light to dark mode updateDepth(GroupParamItem) will insert what I think is the
+        # invisible root node. Most of the time, in the example app the title for this node
+        # will be empty, occasionally its named params.
```

Comment:
"it is name 'params'"? 
Since the comment is long already, maybe include what actually happens?
"In this case, we hide it again by setting sizeHint to zero."? (Is that what happens?)

Reply:
>"it is name 'params'"?

changed it to `occasionally the title is 'params'.`, if that helps to be more clear. 

> "In this case, we hide it again by setting sizeHint to zero."? (Is that what happens?)

Not exactly, it appears to be hidden by default, unless given a size hint by the code below. I'm making a small change to the code so its easier to tell whats happening. 

```py
        title = self.param.title()
        if not title or title == 'params':
            return
        self.setText(0, title)
        fm = QtGui.QFontMetrics(self.font(0))
        textFlags = QtCore.Qt.TextFlag.TextSingleLine
        size = fm.size(textFlags, self.text(0))
        size.setHeight(int(size.height() * 1.35))
        size.setWidth(int(size.width() * 1.15))
        self.setSizeHint(0, size)
```

Reply:
How about just sticking a comment on the setSizeHint line?
`# This makes sure that the label remains visible`  
...or invisible, because I still don't know what this is actually trying to do :)

Reply:
The only thing I've changed here is in this method is:

```py
        # This makes sure that items with no title or the title 'params' remain invisible
        
        title = self.param.title()
        if not title or title == 'params':
            return
```
If there isn't a title or the title is ''params'` then we don't want to run the code below to give it a size. It seems to default to 0,0 width/height, but I didn't look very deeply into it because simply checking that the title wasn't empty did what we wanted. 

> How about just sticking a comment on the setSizeHint line?
This isn't my code, but the code on the size hint line sets a size hint of `size` for column 0. The size is set to the width/height *1.35 of a single line of whichever font self.font(0) returns

Reply:
I maybe missed this bit then
--> `# This makes sure that items with no title or the title 'params' remain invisible`
That was the thing that wasn't clear to me.

Sorry! It just felt weird that there was a three line comment that seemed to build up to something and then didn't tell me what it was actually doing in the end :)